src/classes/TDTM_Runnable.cls

```diff
@@ -148,10 +148,18 @@ global abstract class TDTM_Runnable {
         * @return void
         */
         public void groupByType() {
-            objectsToInsert = groupSObjects(objectsToInsert);
-            objectsToUpdate = groupSObjects(objectsToUpdate);
-            objectsToDelete = groupSObjects(objectsToDelete);
-            objectsToUndelete = groupSObjects(objectsToUndelete);
+            if (!objectsToInsert.isEmpty()) {
```

Comment:
I recommend using the null safe operator in these if blocks just in case
```suggestion
            if (objectsToInsert?.isEmpty() == false) {
```

Reply:
These variables are instantiated as empty lists.  Is the null safe operator necessary?

Reply:
If you're confident the null check is never required, then no.

Reply:
Just double checked - they definitely can be set to null, and the fields are exposed globally instead of accessed using setters so the null safe operator is the way to go!  Just pushed that up in a commit.  Thank you for the catch and suggestion.